### DISCUSSION TITLE
Checkout code fails when it runs: lerna run test --since master

### DISCUSSION THREAD

**Discussion by mateusmarquezini** [community/community#25192] [Actions]
<p>Hey folks,</p>
<p>My checkout code action on GitHub Workflow is failing when it tries to run the following command: </p>
<pre><code>yarn lerna run test --since master -- --coverage
</code></pre>
<p>This is my action step setup:</p>
<pre><code>steps:
  - uses: actions/checkout@v2
</code></pre>
<p>The full error is below:</p>
<pre><code>$ cross-env CI=true lerna run test --since master -- --coverage
93
lerna notice cli v3.20.2
94
lerna info ci enabled
95
lerna notice filter changed since "master"
96
lerna info Looking for changed packages since master
97
lerna ERR! Error: Command failed: git diff --name-only master -- packages/app
98
lerna ERR! fatal: bad revision 'master'
99
lerna ERR!
100
lerna ERR!
101
lerna ERR! at makeError (/home/runner/work/backstage/backstage/node_modules/execa/index.js:174:9)
102
lerna ERR! at Function.module.exports.sync (/home/runner/work/backstage/backstage/node_modules/execa/index.js:338:15)
103
lerna ERR! at Object.execSync (/home/runner/work/backstage/backstage/node_modules/@lerna/child-process/index.js:26:16)
104
lerna ERR! at diffSinceIn (/home/runner/work/backstage/backstage/node_modules/@lerna/collect-updates/lib/make-diff-predicate.js:63:23)
105
lerna ERR! at hasDiffSinceThatIsntIgnored (/home/runner/work/backstage/backstage/node_modules/@lerna/collect-updates/lib/make-diff-predicate.js:27:18)
106
lerna ERR! at isCandidate (/home/runner/work/backstage/backstage/node_modules/@lerna/collect-updates/collect-updates.js:88:77)
107
lerna ERR! at /home/runner/work/backstage/backstage/node_modules/@lerna/collect-updates/lib/collect-packages.js:11:9
108
lerna ERR! at Map.forEach (&lt;anonymous&gt;)
109
lerna ERR! at collectPackages (/home/runner/work/backstage/backstage/node_modules/@lerna/collect-updates/lib/collect-packages.js:10:12)
110
lerna ERR! lerna Command failed: git diff --name-only master -- packages/app
111
lerna ERR! lerna fatal: bad revision 'master'
112
lerna ERR! lerna
113
</code></pre>
<p>I think the checkout code action makes pull from a branch with a different name from master, and when lerna runs using the flag <strong>–since</strong> , it runs git diff --name-only master – packages/app looking for the branch master and doesn’t find it. I don’t know.</p>
<p>Can anybody help me, please? Any kind of help will be very appreciated.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/mateusmarquezini">@mateusmarquezini</a> ,</p>
<p>I noticed that you execute the checkout action without setting any inputs, when you use  <strong>checkout@v2</strong> in your workflow, the below two points need you pay attention to:</p>
<ol>
<li>It fetches only a single commit by default</li>
</ol>
<p>2) No longer detached HEAD when checking out a branch</p>
<p>That means, by default the action only check out the lasted commit ( <strong>github.sha</strong> ) from the branch ref ( <strong>github.ref</strong> ) when the workflow is triggered. So no have other commits or branches be checked out for comparison ( <strong>git diff</strong> ).</p>
<p>If you want to compare between different commits in the branch ref, you need to use the input parameter  <strong>fetch-depth</strong> to specify the number of commits to fetch. </p>
<p>For example:</p>
<pre><code>- name: Checkout
  uses: actions/checkout@v2
  with:
    fetch-depth: 2 # Number of commits to fetch. 0 indicates all history. Default: 1
</code></pre>
<p>If you want to compare between different branches in the repository, you need to execute extra commands to checkout other branches.</p>
<p>For exapmle:</p>
<pre><code>- name: Checkout
uses: actions/checkout@v2

- name: Fetch all history for all tags and branches
  run: git fetch
</code></pre>
<p>More details, you can reference teh docs about <strong>checkout@v2</strong> action here: <a href="https://github.com/actions/checkout/tree/v2#checkout-v2" rel="nofollow noopener">https://github.com/actions/checkout/tree/v2#checkout-v2</a></p>

**Comment by mateusmarquezini** [comment]
<p><a class="mention" href="/brightran">@brightran</a> Using git fetch command to fetch all branches in a new step solved my problem.</p>
<p>Thank you so much!</p>
